### PR TITLE
initial commit

### PR DESCRIPTION
**Pull request by resdenia**
Initial Commit.
WIP:
- Connect `config form` populate data to backend.
- Add status after starting locally.
- Playwright saves the last page in HAR. Need to add an option to have an array of the HAR data for each page.
- Add sessionId to each log.
- README

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,73 @@
 # synthetic-scripting
+
```
Comment (yotamloe) [inline README.md:2]: Before I start reviewing the code, its hard to tell what this integration is used for. Can you add a short description to the readme of what this integration actually does?

File: service/lambdaFunction/logger.js [hunk 2]
```diff
@@ -0,0 +1,11 @@
+const loggerGenerator = (token) => {
+    return require('logzio-nodejs').createLogger({
+        token: token,
+        protocol: 'https',
+        host: 'listener.logz.io',
```
Comment (yotamloe) [inline service/lambdaFunction/logger.js]: What happens if the customer account is not in us? looks like the listener address is static

File: index.js [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+app.use(bodyParser.urlencoded({ extended: false }));
+app.use(express.static(path.join(__dirname, 'public')));
+app.set('views', __dirname + '/views');
+app.engine('html', require('ejs').renderFile);
+
+app.use(mainRoutes);
+app.use('/api', creatorRoutes);
+const port = process.env.PORT || 8080;
+app.listen(port, () => {
+    console.log(`helloworld: listening on port ${port}`);
```
Comment (yotamloe) [inline index.js]: why `helloworld` and not the name of the app?

File: README.md [hunk 4]
```diff
@@ -1 +1,73 @@
...
+`docker run -p 8080:8080 -d --name synthetic-script logzio/synthetic-script:latest`
+
+Docker image will run Node js mini app. Go to http://localhost:8080, on you Browser you will see UI that help you to create Connection between your app and logz.io.
+
+## Define script for running test
+
+1. On a First tab `General Settings` you will see Please fill all reqiured fields.
+
+![UI first screen](assets/first-screen.png)
```
Comment (yotamloe) [inline README.md:23]: Should the user press accept configs before testing locally? it is not clear

File: utils/cloudWatchEvent.js [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+const { EventBridgeClient } = require('@aws-sdk/client-eventbridge');
+
+exports.cloudWatchEvent = async (
+    name,
+    rangeTime,
+    region,
+    accessKey,
+    secretKey,
+    iamRoleArnEvent,
+    accountId,
```
Comment (yotamloe) [inline utils/cloudWatchEvent.js]: Why does the user need to provide the `account id` if we have `accessKey` and `secretKey` cant we just get the account id from the AWS sdk?

File: utils/lambda-creator.js [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+            ? functionName.split(' ').join('-')
+            : functionName;
+    const params = {
+        Code: {
+            S3Bucket: bucketName, // BUCKET_NAME
+            S3Key: `${nameZip}.zip`, // ZIP_FILE_NAME
+        },
+        FunctionName: functionName || LAMBDA_FUNCTION_NAME,
+        Handler: 'index.handler',
+        Role: iamRoleArn, // IAM_ROLE_ARN; e.g., arn:aws:iam::650138640062:role/v3-lambda-tutorial-lambda-role
```
Comment (yotamloe) [inline utils/lambda-creator.js]: Suggestion: to improve user experience can we create the lambda execution role using AWS sdk, instead of making the user manually create it and copy paste the arn?

File: utils/cloudWatchEvent.js [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+            Targets: [
+                {
+                    Arn: `arn:aws:lambda:${region}:${accountId}:function:${name}`, //LAMBDA_FUNCTION_ARN
+                    Id: 'myCloudWatchEventsTarget',
+                },
+            ],
+        };
+        const paramsRule = {
+            Name: CLOUDWATCH_EVENT,
+            RoleArn: iamRoleArnEvent,
```
Comment (yotamloe) [inline utils/cloudWatchEvent.js]: Suggestion: to improve user experience can we create the `iamRoleArnEvent` using AWS SDK, instead of making the user manually create it and copy-paste the arn?

File: README.md [hunk 8]
```diff
@@ -1 +1,73 @@
...
+| Aws Access Key (Required)             | Your AWS access key ID.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               |
+| Aws Secret Key (Required)             | Your AWS secret key ID.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               |
```
Comment (yotamloe) [inline README.md]: We need to tell our users what are the minimum permissions required for the iam user, for this integration to work

**Comment by yotamloe** [comment]
Suggestion: It's hard for the user to see what's going on in the program when you press buttons like `set env variables` or `test locally` is there a relatively easy way for you to implement a log stream console that shows the program logs in the web UI?

File: utils/update-function-local.js [hunk 9]
```diff
@@ -0,0 +1,50 @@
...
+const path = require('path');
+const { startFileLocally, endFileLocally } = require('../helper/index-config');
+
+const readWriteAsync = async (code, filePath) => {
+    try {
+        const fileStarts = startFileLocally.split('\n');
+        const IDENTIFIER_CODE = `///////////////////////////////////`;
+        const fileEnds = endFileLocally.split('\n');
+        const extractCode = code.split(IDENTIFIER_CODE);
+        console.log(extractCode.length);
```
Comment (yotamloe) [inline utils/update-function-local.js]: Why do we need to log the code length?

File: helper/index-config.js [hunk 10]
```diff
@@ -0,0 +1,72 @@
...
+				recordHar: {
+					path: './capture-hars/example.har',
+					mode: 'full',
+					content: 'omit',
+				},
+			});
+			page = await context.newPage();
+	`,
+    endFileLocally: `    } catch (error) {
+		console.log('is here?', error);
```
Comment (yotamloe) [inline helper/index-config.js]: why do we need to log `is here?` before the error?

**Review by yotamloe — CHANGES_REQUESTED** [review]
Impressive project!! Please see my comments

File: README.md [hunk 11]
```diff
@@ -1 +1,73 @@
 # synthetic-scripting
+
+## Run Application
+
+1. Pull the Docker image
```
Comment (nico-shishkin) [inline README.md]:
```suggestion
1. Pull the Docker image.
```

File: README.md [hunk 12]
```diff
@@ -1 +1,73 @@
...
+
+`docker pull logzio/synthetic-script:latest`
+
+2. Run the docker
```
Comment (nico-shishkin) [inline README.md]:
```suggestion
2. Run the Docker container.
```

File: README.md [hunk 13]
```diff
@@ -1 +1,73 @@
...
+
+`docker run -p 8080:8080 -d --name synthetic-script logzio/synthetic-script:latest`
+
+Docker image will run Node js mini app. Go to http://localhost:8080, on you Browser you will see UI that help you to create Connection between your app and logz.io.
```
Comment (nico-shishkin) [inline README.md]:
```suggestion
Docker image will run a Node.js mini app. Open http://localhost:8080 in your Browser to access the UI that helps you establish connection between your app and Logz.io.
```

File: README.md [hunk 14]
```diff
@@ -1 +1,73 @@
...
+
+## Define script for running test
```
Comment (nico-shishkin) [inline README.md]:
```suggestion
## Define a script for running the test
```

File: README.md [hunk 15]
```diff
@@ -1 +1,73 @@
...
+
+1. On a First tab `General Settings` you will see Please fill all reqiured fields.
```
Comment (nico-shishkin) [inline README.md]:
```suggestion
1. In tab `General Settings`, fill in all the required fields.
```

File: README.md [hunk 16]
```diff
@@ -1 +1,73 @@
...
+
+![UI first screen](assets/first-screen.png)
+
+| Field                                                    | Description                                                                                                               |
+| -------------------------------------------------------- | ------------------------------------------------------------------------------------------------------------------------- |
+| Name of the Lambda (Required)                            | Lambda function name what will be created and it also will be a name of the text to represent it inside logz.io Dashboard |
+| Description (Optional)                                   | Lambda function description what will be create                                                                           |
+| Logz.io shipping Token (Required)                        | Your Logz.io metrics account token.                                                                                       |
+| Range time for send data to the logs (Default: 1 minute) | Define a time in a minutes to run a lambda function (using cloudBridge event)                                             |
```
Comment (nico-shishkin) [inline README.md]:
```suggestion
| Time range for sending logs (Default: 1 minute) | Define the range in a minutes to run a Lambda function (using cloudBridge event)                                             |
```

File: README.md [hunk 17]
```diff
@@ -1 +1,73 @@
...
+
+1. On a First tab `General Settings` you will see Please fill all reqiured fields.
+
+![UI first screen](assets/first-screen.png)
+
+| Field                                                    | Description                                                                                                               |
+| -------------------------------------------------------- | ------------------------------------------------------------------------------------------------------------------------- |
+| Name of the Lambda (Required)                            | Lambda function name what will be created and it also will be a name of the text to represent it inside logz.io Dashboard |
+| Description (Optional)                                   | Lambda function description what will be create                                                                           |
+| Logz.io shipping Token (Required)                        | Your Logz.io metrics account token.                                                                                       |
```
Comment (nico-shishkin) [inline README.md]:
```suggestion
| Logz.io shipping Token (Required)                        | Your Logz.io logs shipping token.                                                                                       |
```

File: README.md [hunk 18]
```diff
@@ -1 +1,73 @@
...
+
+## Define script for running test
+
+1. On a First tab `General Settings` you will see Please fill all reqiured fields.
+
+![UI first screen](assets/first-screen.png)
+
+| Field                                                    | Description                                                                                                               |
+| -------------------------------------------------------- | ------------------------------------------------------------------------------------------------------------------------- |
+| Name of the Lambda (Required)                            | Lambda function name what will be created and it also will be a name of the text to represent it inside logz.io Dashboard |
```
Comment (nico-shishkin) [inline README.md]:
```suggestion
| Name of the Lambda (Required)                            | Lambda function name what will be created. This name will also be used to identify the Lambda function in Logz.io Dashboard |
```

File: README.md [hunk 19]
```diff
@@ -1 +1,73 @@
...
+| Description (Optional)                                   | Lambda function description what will be create                                                                           |
```
Comment (nico-shishkin) [inline README.md]:
```suggestion
| Description (Optional)                                   | Lambda function description                                                                           |
```

File: README.md [hunk 20]
```diff
@@ -1 +1,73 @@
...
+| Aws Secret Key (Required)             | Your AWS secret key ID.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               |
+| Bucket Name (Required)                | The bucket name to which the PUT action was initiated. When using this action with an access point, you must direct requests to the access point hostname. The access point hostname takes the form AccessPointName-AccountId.s3-accesspoint.Region.amazonaws.com. When using this action with an access point through the Amazon Web Services SDKs, you provide the access point ARN in place of the bucket name. For more information about access point ARNs, see Using access points in the Amazon S3 User Guide. |
```
Comment (nico-shishkin) [inline README.md]:
```suggestion
| Bucket Name (Required)                | The bucket name to which the PUT action is initiated. When using this action with an access point, you must direct requests to the access point hostname. The access point hostname takes the form AccessPointName-AccountId.s3-accesspoint.Region.amazonaws.com. When using this action with an access point through the Amazon Web Services SDKs, you need to provide the access point ARN instead of the bucket name. For more information about access point ARNs, see Using access points in the Amazon S3 User Guide. |
```

File: README.md [hunk 21]
```diff
@@ -1 +1,73 @@
...
+| Aws Region (Required)                 | Your region to send service requests to.                                                                                                                                                                                                                                                                                                                                                                                                                                                                              |
```
Comment (nico-shishkin) [inline README.md]:
```suggestion
| Aws Region (Required)                 | Your AWS region to send service requests to.                                                                                                                                                                                                                                                                                                                                                                                                                                                                              |
```

File: README.md [hunk 22]
```diff
@@ -1 +1,73 @@
...
+| AWS IAM Role Lambda Create (Required) | The Amazon Resource Name (ARN) of the function's execution role.                                                                                                                                                                                                                                                                                                                                                                                                                                                      |
+| AWS IAM Role Event Put (Required)     | Add Rule for put event to lambda function                                                                                                                                                                                                                                                                                                                                                                                                                                                                             |
```
Comment (nico-shishkin) [inline README.md]:
```suggestion
| AWS IAM Role Event Put (Required)     | Add a rule for PUT events to the Lambda function.                                                                                                                                                                                                                                                                                                                                                                                                                                                                       |
```

File: README.md [hunk 23]
```diff
@@ -1 +1,73 @@
...
+
+2. On a Third tab `Enviroment Variables` you will see.
+
+![UI third screen](assets/third-screen.png)
+
+You can define here env variable that can be attach for your lambda function and you can use it in Code Editor as `process.env.KEY` where Key you can define in that section and it will be prepresent inside you lambda function
```
Comment (nico-shishkin) [inline README.md]:
```suggestion
Here you can define the environment variable that can be attached to your Lambda function. You can use it in the Code Editor as `process.env.KEY`. You can define the KEY parameter, which will be present in the Lambda function
```

Comment (nico-shishkin) [inline README.md]: What will the KEY parameter do exactly?

File: README.md [hunk 24]
```diff
@@ -1 +1,73 @@
...
+
+3. On A Fourth tab `Code Editor` you will see.
+
+![UI four screen](assets/four-screen.png)
+
+You can define code here for your test using [playwright.dev](https://playwright.dev). But please put your code specific between comments
```
Comment (nico-shishkin) [inline README.md]:
```suggestion
Here you can define the code for your test using [playwright.dev](https://playwright.dev). Put your code between the comments.
```

File: README.md [hunk 25]
```diff
@@ -1 +1,73 @@
...
+
+4. After set all values and define code of test what you want to run.
```
Comment (nico-shishkin) [inline README.md]:
```suggestion
4. After setting all the values and defining the test code,
```

File: README.md [hunk 26]
```diff
@@ -1 +1,73 @@
...
+   You can you can click `Test locally` if everything works you will see metrics from one your test.
```
Comment (nico-shishkin) [inline README.md]:
```suggestion
click `Test locally` to check if everything works ad the logs are generated.
```

File: README.md [hunk 27]
```diff
@@ -1 +1,73 @@
...
+   For Deploy your lambda function you can click `Deploy to the cloud`. You will see a status of all steps what is running behind. if Everything is good you need to see
```
Comment (nico-shishkin) [inline README.md]:
```suggestion
   To deploy the Lambda function, click `Deploy to the cloud`. You will see status of all the steps running in the background. The following message appears when the checks are completed:
```

File: service/lambdaFunction/logger.js (hunk 2 above)
Comment (resdenia) [reply service/lambdaFunction/logger.js]: Adding option to manage it from UI

File: utils/cloudWatchEvent.js (hunk 5 above)
Comment (resdenia) [reply utils/cloudWatchEvent.js]: Yes will add a function to receive account id

File: utils/update-function-local.js (hunk 9 above)
Comment (resdenia) [reply utils/update-function-local.js]: Will remove it

File: helper/index-config.js (hunk 10 above)
Comment (resdenia) [reply helper/index-config.js]: Will remove it

File: utils/cloudWatchEvent.js [hunk 28]
```diff
@@ -0,0 +1,100 @@
...
+
+        if (iamRole.err) {
+            throw iamRole.err;
+        }
+
+        const paramsRule = {
+            Name: `${rangeTime}-${PUT_RULE_ROLE_NAME}`,
+            RoleArn: iamRole.arn,
+            ScheduleExpression: `rate(${rangeTime} minute${
+                parseInt(rangeTime) === 1 ? '' : 's'
```
Comment (Doron-Bargo) [inline utils/cloudWatchEvent.js]: Please add the radix parameter for preventing the unintended conversion of a string to a number of a different base

File: utils/AIM-role.js [hunk 29]
```diff
@@ -0,0 +1,101 @@
...
+        secretAccessKey: secretKey,
+    });
+    const iam = new AWS.IAM();
+    try {
+        const getIfRoleExists = await isRoleExists(iam, roleName);
+        if (getIfRoleExists.Role) {
+            return { arn: getIfRoleExists.Role.Arn };
+        }
+    } catch (err) {
+        logger("Role Doesn't exist process to create");
```
Comment (Doron-Bargo) [inline utils/AIM-role.js]: What is the meaning of this log?

**Review by Doron-Bargo — CHANGES_REQUESTED** [review]
Looking good
Small changes

File: utils/AIM-role.js (hunk 29 above)
Comment (resdenia) [reply utils/AIM-role.js]: Update logger message. It's for internal use to see that we are starting to create new role.

File: utils/cloudWatchEvent.js (hunk 28 above)
Comment (resdenia) [reply utils/cloudWatchEvent.js]: Add

File: utils/AIM-role.js (hunk 29 above)
Comment (Doron-Bargo) [reply utils/AIM-role.js]: Can you rephrase the message in the log - Its not clear

**Review by Doron-Bargo — APPROVED** [review]
lgtm

**Review by yotamloe — APPROVED** [review]
